MetaMask/metamask-extension
# Replace incoming with external transactions

Fixes: #505 #7452

Explanation:  

1.  `IncomingTransactionsController` has been replaced with `ExternalTransactionsController`
2.  `showIncomingTransactions` feature flag has been changed to `showExternalTransactions`
3. `ExternalTransactionsController` now does not filter out outgoing transactions
4.  New transactions type `SENT` has been added, wich is described as "External Send" transaction
5.  `_normalizeTxFromEtherscan` sets the new transaction type `SENT` when address of the sender is the same as the one passed as the parameter

<img width="359" alt="Screenshot 2021-04-20 at 12 51 58" src="https://user-images.githubusercontent.com/5708018/115512474-53c80e00-a282-11eb-97ca-9334dfd870e1.png">

Manual testing steps:  
  - Import account using private key
  - Use the same private key to import it to other instance of MM or any other wallet
  - Send transaction using 2nd wallet
  - Wait for the block request in 1st MM instance
  - Check if "External Send" transaction will show up
